api/api/utils/generate_teams_data_loader.py

```diff
@@ -1,40 +1,76 @@
-from typing import Dict, Any, List
-from dataclasses import dataclass, field
+from typing import Dict, Any, List, Optional
+from dataclasses import dataclass
+
+from schema import Schema
 
 from api.ai.new.interfaces.algorithm_options import AlgorithmOptions
 from api.ai.new.interfaces.team_generation_options import TeamGenerationOptions
 from api.models.enums import AlgorithmType
-from api.models.project import Project
 from api.models.student import Student
+from api.api.utils.algorithm_option import get_algorithm_option_class
 
 
 @dataclass
 class GenerateTeamsInputData:
     students: List[Student]
     algorithm_type: AlgorithmType
-    algorithmOptions: AlgorithmOptions
+    algorithm_options: AlgorithmOptions
     team_generation_options: TeamGenerationOptions
-    projects: List[Project] = field(default_factory=list)
 
 
 class GenerateTeamsDataLoader:
     def __init__(self, validated_data: Dict[str, Any]):
         self.data = validated_data
 
     def load(self) -> GenerateTeamsInputData:
```

Comment:
In general, if the loader is loading a non-primitive, it needs to convert that nested dict, list, or value into the correct enum or object type